keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/DoubleVertex.java

```diff
@@ -335,11 +338,25 @@ public PartialDerivatives forwardModeAutoDifferentiation(Map<Vertex, PartialDeri
         }
     }
 
+    @Override
+    public void setValue(KeanuSavedBayesNet.VertexValue valueBuf) {
+        if (valueBuf.getValueTypeCase() != KeanuSavedBayesNet.VertexValue.ValueTypeCase.DOUBLEVAL) {
+            throw new IllegalArgumentException("Non Double Value specified for Double Vertex");
```

Comment:
This pair of lines appears twice. If you encapsulate it as a static method somewhere, you'll be able to reuse it many times (covering all the different `ValueTypeCase`s)

Reply:
Refactored in a slightly different way.